mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -252,7 +250,7 @@ class LinearRegression @Since("1.3.0") (@Since("1.3.0") override val uid: String
           model,
           Array(0D),
           Array(0D))
-        return copyValues(model.setSummary(trainingSummary))
+        return model.setSummary(trainingSummary)
```

Comment:
@jkbradley It does not necessary to ```setParent``` at here, because we have done it at ```Predictor.fit```
```
override def fit(dataset: Dataset[_]): M = {
    // This handles a few items such as schema validation.
    // Developers only need to implement train().
    transformSchema(dataset.schema, logging = true)
    copyValues(train(dataset).setParent(this))
  }
```